src/WrappedVault.sol

```diff
@@ -602,8 +616,8 @@ contract WrappedVault is Owned, ERC20, IWrappedVault {
     }
 
     /// @inheritdoc IWrappedVault
-    function maxWithdraw(address) external view returns (uint256 maxAssets) {
-        maxAssets = VAULT.maxWithdraw(address(this));
+    function maxWithdraw(address owner) external view returns (uint256 maxAssets) {
+        return SoladyMath.min(VAULT.previewWithdraw(balanceOf[owner]), VAULT.maxWithdraw(address(this)));
```

Comment:
The first fix that modified `maxMint` and `maxDeposit` to pass on the `receiver` parameter to the underlying vault was the right one, if we follow the standard.

Note that the [ethereum.org](https://ethereum.org/en/developers/docs/standards/tokens/erc-4626/#maxdeposit) description of the standard has an error for `maxWithdraw` and `maxRedeem`. It's not "by the receiver":

<img width="860" alt="image" src="https://github.com/user-attachments/assets/9383ef82-9a53-44ec-9b0b-14507f6fab1b">

It's ["for the receiver"](https://github.com/ethereum/ercs/blob/master/ERCS/erc-4626.md#maxdeposit):

<img width="906" alt="image" src="https://github.com/user-attachments/assets/7d96bdc1-41d9-4782-8919-528e79986725">

The idea is to allow enabling blocklists at the vault level. If you are going to restrict holding vault shares by certain addresses, you would stop it both at transfer and deposit/mint. The `maxDeposit/maxMint` should reflect that.

It is a bit arbitrary, because blocklists could also be implemented on the function caller, and not just the token receiver, but that's what it is.

On limiting the return of `maxRedeem/maxWithdraw` to the minimum of the wrapped vault or wrapped vault holder is something I don't understand. The wrapped vault holder doesn't hold any shares of the underlying vault, so how would that work?

Reply:
Re this:
>On limiting the return of maxRedeem/maxWithdraw to the minimum of the wrapped vault or wrapped vault holder is something I don't understand. The wrapped vault holder doesn't hold any shares of the underlying vault, so how would that work?

My understanding is that since shares in the wrapped vault are assigned 1:1 with the shares created in the underlying vault when a deposit occurs, the user's `balanceOf` is the maximum number of underlying vault shares they are entitled to redeem.

Reply:
However, it's only correct in the case of `maxRedeem` as `previewWithdraw` expects an argument that is assets, not shares.

Reply:
`maxWithdraw` needs to do something like `VAULT.previewWithdraw(VAULT.convertToAssets(balanceOf[owner]))` for the first arg to the `min` instead.